dev/PowerNotifications/PowerNotifications.cpp

```diff
@@ -0,0 +1,316 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <pch.h>
+#include <future>
+#include <PowerNotifications.h>
+#include <frameworkudk\PowerNotificationsPal.h>
+#include <PowerManager.g.cpp>
+#include <powrprof.h>
+
+namespace winrt::Microsoft::ProjectReunion::implementation
+{
+
+    // EnergySaverStatus Functions
+    EventType& EnergySaverStatus_Event()
+    {
+        return make_self<factory_implementation::PowerManager>()->m_energySaverStatusChangedEvent;
+    }
+
+    void EnergySaverStatus_Register()
+    {
+        check_hresult(PowerNotifications_RegisterEnergySaverStatusChangedListener(
```

Comment:
what does check_hresult do?  Isn't that legacy CppWinRT stuff?  THROW_IF_FAILED would be better here and more expressive.

Reply:
Not sure about legacy but THROW_IF_FAILED  is better; agreed.